src/algebra/module/submodule.lean

```diff
@@ -199,6 +199,45 @@ instance : add_comm_group p :=
 
 end add_comm_group
 
+section ordered_monoid
+
+variables [semiring R]
+
+/-- A submodule of a `ordered_add_comm_monoid` is a `ordered_add_comm_monoid`. -/
+instance {M} [ordered_add_comm_monoid M] [semimodule R M] (S : submodule R M) :
+  ordered_add_comm_monoid S :=
+subtype.coe_injective.ordered_add_comm_monoid coe rfl (λ _ _, rfl)
+
+/-- A submodule of a `ordered_cancel_add_comm_monoid` is a `ordered_cancel_add_comm_monoid`. -/
+instance {M} [ordered_cancel_add_comm_monoid M] [semimodule R M] (S : submodule R M) :
+  ordered_cancel_add_comm_monoid S :=
+subtype.coe_injective.ordered_cancel_add_comm_monoid coe rfl (λ _ _, rfl)
+
+/-- A submodule of a `linear_ordered_cancel_add_comm_monoid` is a
+`linear_ordered_cancel_add_comm_monoid`. -/
+instance {M} [linear_ordered_cancel_add_comm_monoid M] [semimodule R M] (S : submodule R M) :
+  linear_ordered_cancel_add_comm_monoid S :=
+subtype.coe_injective.linear_ordered_cancel_add_comm_monoid coe rfl (λ _ _, rfl)
```

Comment:
Do you think we should give these explicit names to match how its done for `subgroup`?

Reply:
I am not sure whether I am the right person to ask: I do not really know what you do with the name of an instance, or how you can use an instance with no name.  Could it hurt to give these names?

Reply:
It only hurts the line length. Names are rarely used for instances.

Reply:
Ok, I am glad that you added the names, since, for me, they help when figuring out what is going on, even if they never get used anywhere else!